File: crawler/src/models/lease-contract.ts
import { DataTypes } from "sequelize";
import sequelize from "../database/database";
import RenterProfile from "./renter-profile";
import Hostel from "./hostel";

const LeaseContract = sequelize.define("LeaseContract", {
    id: {
        type: DataTypes.INTEGER,
        autoIncrement: true,
        primaryKey: true,
    },
    renterId: {
        type: DataTypes.INTEGER,
        references: {
            model: RenterProfile,
            key: "id",
        },
    },
    hostelId: {
        type: DataTypes.INTEGER,
        references: {
            model: Hostel,
            key: "id",
        },
    },
    startDate: {
        type: DataTypes.DATE,
    },
    endDate: {
        type: DataTypes.DATE,
    },
    deposit: {
        type: DataTypes.INTEGER,
    },
    monthlyRent: {
        type: DataTypes.INTEGER,
    },
    status: {
        type: DataTypes.STRING,
        values: ['active', 'inactive'],
        defaultValue: 'active',
    },
}, {
    tableName: 'lease-contract',
    timestamps: true,
    underscored: true,
});

export default LeaseContract;

File: main-project/Models/LeaseContract.cs
namespace main_project.Models;

public class LeaseContract
{
    public int Id { get; set; }
    public int RenterId { get; set; }
    public int HostelId { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public int Deposit { get; set; }
    public int MonthlyRent { get; set; }
    public string Status { get; set; } = "active";
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public virtual RenterProfile? Renter { get; set; }
    public virtual Hostel? Hostel { get; set; }

    public static IQueryable<LeaseContract> Query()
    {
        DatabaseContext db = new DatabaseContext();
        return db.leaseContracts;
    }
}